### ISSUE TITLE
Use nano's for internal time tracking rather than milliseconds

### ISSUE THREAD

**Issue by jentfoo** [threadly/threadly#178]
With this commit: https://github.com/threadly/threadly/commit/7849297b1c1047a0d6d79cffbcdd336b41cd6909 we are not able to use nanos rather than milliseconds.

Originally I avoided using nanosecond because I did not think it was worth it with system clock resolution/accuracy.  I have now convinced myself we should switch to nanoseconds for our internal time tracking (which is what java.util.concurrent does as well).  This would allow a little math reduction, but more importantly if a system has a more accurate clock we are able to take advantage of it.

Basically I just convinced myself that the human complexity of nanoseconds, is worth it.

**Comment by jentfoo** [comment]
I have a prototype branch for this, and decided it's not worth going forward with right now (until someone convinces me otherwise).  here is the reasoning:

Advantages....
* It moves the millis -> nano math into submission time rather than execution time
* It can potentially make the execution for scheduled and recurring tasks more accurate if the system clock is more accurate

Reasons not to go forward with this...
* Even though there is less math at execution time, it does increase the needed math in general (Spreads it out since all our API's use millis)
* We now have to deal with overflows a lot more in cases where Long.MAX_VALUE and other similar high values maybe be provided
* The benchmarks show no statistically significant performance improvement at all
* The execution time improvement would only occur for scheduled and recurring tasks.  I don't believe these types of tasks should ever expect nanosecond execution time accuracy.  Since execute tasks time calculation is arbitrary (they are immediately available for execution), there is no advantage for them.
